2 little hack for functionnalities for developers

Angra974 opened the pull request:
1/ Don't stop the application after a successful installation.
     => Normal behavior:
           After a successful installation, if we need to do another one, we need to kill the process and load the application again.
     => Wanted behavior and process
           When an installation succeded, a new one can be done and we go for asking a new name for the next project
                Process:
                       Added a constant REPEATED ( in /src/constants.ts ), the default set to false.
                       Set to false ( default ), we will have the actual behavior of the application
                       Set to true (change made by the developer ), the application repeats installations till the process is killed

1/ Don't install dependencies during the installation.
     => Normal behavior:
           While running the application, it installs all the dependencies for this application.
     => Wanted behavior and process
           While running the application, the dependencies installation can be shunted.
                Process:
                       Added a constant INSTALL_DEP ( in /src/constants.ts ), the default set to true.
                       Set to true ( default ), we will have the actual behavior of the application
                       Set to false ( change made by the developer), the application won't install the dependencies and the developer
will be able to check the result and do others process without waiting for the application to finish.

Utilization in development: 
In my case actually, for just the vanilla + vanilla ts installations and checking/verifications, I need to do about 10 installations while changing some parameters to check if all I've made are ok. 
So I can bypass all the 10 installations of dependencies and the need to start the application from start.

Utilization in production:
  Nothing changed from the normal behavior.

Angra974 committed:
Create local-ci.yml

MyCi

Angra974 committed:
Delete local-ci.yml

Just deleted

Angra974 committed:
Added 2 developer friendly constants

AndrejJurkin reviewed (changes requested):
wow, that's a great PR! thank you so much for this.

let's move these variables from constant.ts to an .env file,
this way every developer would be able to set their environment without changing the files in source control.

it would be nice to also add .env.example file which will be included in git, so other people know what options do we have.

Angra974 commented:
I've sent it like this because I don't want to change too many things at the time so you can guide the way how you will be coded.

I've made more changes but try to get them in order and not have all of them mixed together.

Angra974 committed:
Merge branch 'main' into feat/developerFriendly

Angra974 committed:
Update index.ts

Needed to add await before main() to load the application again due to error lint error concerning promise.
